### ISSUE TITLE
Add a change_quantity method

### ISSUE THREAD

**Issue by excid3** [pay-rails/pay#186]
To better support quantities on subscriptions, we should add a method to update them. This would be called in the app anytime the quantity changes (ie. if you're pricing by users, you would call this any time you add or remove a user).

Right now, only Stripe is supported, but we should figure out how to make Braintree work similarly for quantities.

**Comment by guillaumebriday** [comment]
That would be nice!

Did you start something or can I tried to help from scratch?

**Comment by garethfr** [comment]
looking quickly at [subscription.rb](https://github.com/pay-rails/pay/blob/master/lib/pay/stripe/subscription.rb), would this be more complicated than adding an `update` method, similar to `cancel`? Something like : 

```
      def update(**options)
        ::Stripe::Subscription.update(processor_id, options, stripe_options)
        pay_subscription.update(options)
      rescue ::Stripe::StripeError => e
        raise Pay::Stripe::Error, e
      end
```

**Comment by excid3** [comment]
It would have to change the correct line item(s) which might be the tricky part.

**Comment by excid3** [comment]
Considering adding a Pay::SubscriptionItem so we can better handle this rather than just storing it as jsonb on the `Pay::Subscription`. The benefit is that we can more easily query a `Pay::SubscriptionItem`.

* Add `SubscriptionItem` model that syncs only with Stripe
* Still set the processor_plan and quantity on the `Pay::Subscription` using the first SubscriptionItem metadata (for backwards compatibility). This would be ignored if someone wants to query subscriptions with multiple Items.
* We can use the `change_quantity` method on `Pay::Subscription` to update either the first subscription item, or the specified `subscription_item_id`

**Comment by excid3** [comment]
Closed in 97f712f58272611bdfc6dbca4ae36dc9f3a34baa
